library/core/src/num/nonzero.rs

```diff
@@ -2214,4 +2227,152 @@ mod verify {
     nonzero_check!(u128, core::num::NonZeroU128, nonzero_check_new_unchecked_for_u128);
     nonzero_check!(usize, core::num::NonZeroUsize, nonzero_check_new_unchecked_for_usize);
+
+
+    macro_rules! nonzero_check_from_mut_unchecked {
+        ($t:ty, $nonzero_type:ty, $harness_name:ident) => {
+            #[kani::proof]
+            pub fn $harness_name() {
+                let mut x: $t = kani::any();
+                kani::assume(x != 0);
+                unsafe {
+                    <$nonzero_type>::from_mut_unchecked(&mut x);
+                }
+            }
+        };
+    }
+    
+    // Generate harnesses for multiple types
+    nonzero_check_from_mut_unchecked!(i8, core::num::NonZeroI8, nonzero_check_from_mut_unchecked_i8);
+    nonzero_check_from_mut_unchecked!(i16, core::num::NonZeroI16, nonzero_check_from_mut_unchecked_i16);
+    nonzero_check_from_mut_unchecked!(i32, core::num::NonZeroI32, nonzero_check_from_mut_unchecked_i32);
+    nonzero_check_from_mut_unchecked!(i64, core::num::NonZeroI64, nonzero_check_from_mut_unchecked_i64);
+    nonzero_check_from_mut_unchecked!(i128, core::num::NonZeroI128, nonzero_check_from_mut_unchecked_i128);
+    nonzero_check_from_mut_unchecked!(isize, core::num::NonZeroIsize, nonzero_check_from_mut_unchecked_isize);
+    nonzero_check_from_mut_unchecked!(u8, core::num::NonZeroU8, nonzero_check_from_mut_unchecked_u8);
+    nonzero_check_from_mut_unchecked!(u16, core::num::NonZeroU16, nonzero_check_from_mut_unchecked_u16);
+    nonzero_check_from_mut_unchecked!(u32, core::num::NonZeroU32, nonzero_check_from_mut_unchecked_u32);
+    nonzero_check_from_mut_unchecked!(u64, core::num::NonZeroU64, nonzero_check_from_mut_unchecked_u64);
+    nonzero_check_from_mut_unchecked!(u128, core::num::NonZeroU128, nonzero_check_from_mut_unchecked_u128);
+    nonzero_check_from_mut_unchecked!(usize, core::num::NonZeroUsize, nonzero_check_from_mut_unchecked_usize);
+                 
+
+
+    macro_rules! check_mul_unchecked_small{
+        ($t:ty, $nonzero_type:ty, $nonzero_check_mul_for:ident) => {
+            #[kani::proof_for_contract(<$t>::unchecked_mul)]
+            pub fn $nonzero_check_mul_for() {
+                let x: $nonzero_type = kani::any();
+                let y: $nonzero_type = kani::any();
+
+                unsafe {
+                    x.unchecked_mul(y);
+                }
+            }
+        };
+    }
+
+    macro_rules! check_mul_unchecked_intervals{
+        ($t:ty, $nonzero_type:ty, $nonzero_check_mul_for:ident, $min:expr, $max:expr) => {
+            #[kani::proof_for_contract(<$t>::unchecked_mul)]
+            pub fn $nonzero_check_mul_for() {
+                let x = kani::any::<$t>();
+            let y = kani::any::<$t>();
+
+            kani::assume(x != 0 && x >= $min && x <= $max);
+            kani::assume(y != 0 && y >= $min && y <= $max);
+
+            let x = <$nonzero_type>::new(x).unwrap();
+            let y = <$nonzero_type>::new(y).unwrap();
```

Comment:
```suggestion
                let x = kani::any::<$nonzero_type>();
                let y = kani::any::<$nonzero_type>();
                let nonzero_min = NonZero::new($min).unwrap();
                let nonzero_max = NonZero::new($max).unwrap();
    
                kani::assume(x >= nonzero_min && x <= nonzero_max);
                kani::assume(y >= nonzero_min && y <= nonzero_max);
```

Reply:
Here's a link to this macro on the [Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0cc6a16affc2c0a25f04a3f07d3463f4)--click "Tools" --> "Expand Macros" then scroll to the bottom of the output. This is a useful way to visualize what your macro actually expands to (i.e., what code you're actually writing).